stdlib/http/src/main/java/org/ballerinalang/net/http/compiler/ResourceSignatureValidator.java

```diff
@@ -70,27 +75,139 @@ public static void validateAnnotation(FunctionNode resourceNode, DiagnosticLog d
-        } else if (count == 1) {
-            for (BLangRecordLiteral.BLangRecordKeyValue keyValue : annVals) {
-                if (((BLangSimpleVarRef) (keyValue.key).expr).variableName
-                        .getValue().equals("webSocketUpgrade")) {
-                    if (annVals.size() > 1) {
-                        dlog.logDiagnostic(Diagnostic.Kind.ERROR, resourceNode.getPosition(),
-                                           "Invalid configurations for WebSocket upgrade resource");
-                    } else if (((BLangRecordLiteral) keyValue.valueExpr).keyValuePairs.size() == 1) {
-                        if (!((BLangSimpleVarRef) (((BLangRecordLiteral) keyValue.valueExpr).keyValuePairs).get(
-                                0).key.expr).variableName.getValue().equals("upgradeService")) {
-                            dlog.logDiagnostic(Diagnostic.Kind.ERROR, resourceNode.getPosition(),
-                                               "An upgradeService need to be specified for the WebSocket upgrade " +
-                                                       "resource");
+        } else if (count == 0) {
+            return;
+        }
+        for (BLangRecordLiteral.BLangRecordKeyValue keyValue : annVals) {
+            switch (getAnnotationFieldKey(keyValue)) {
+                case ANN_CONFIG_ATTR_WEBSOCKET_UPGRADE:
+                    validateWebSocketUpgrade(resourceNode, dlog, annVals, paramSegments, keyValue);
+                    break;
+                case ANN_RESOURCE_ATTR_PATH:
+                    validateResourcePath(dlog, paramSegments, keyValue);
+                    break;
+                case ANN_RESOURCE_ATTR_BODY:
+                    List<? extends SimpleVariableNode> parameters = resourceNode.getParameters();
+                    String bodyFieldValue = keyValue.getValue().toString();
+                    // Data binding param should be placed as the last signature param
+                    String signatureBodyParam = parameters.get(parameters.size() - 1).getName().getValue();
+                    if (bodyFieldValue.isEmpty()) {
+                        dlog.logDiagnostic(Diagnostic.Kind.ERROR, keyValue.getValue().getPosition(),
+                                           "Empty data binding param value");
+
+                    } else if (!signatureBodyParam.equals(bodyFieldValue)) {
+                        dlog.logDiagnostic(Diagnostic.Kind.ERROR, keyValue.getValue().getPosition(),
+                                           "Invalid data binding param in the signature : expected '" +
+                                                   bodyFieldValue + "', but found '" + signatureBodyParam + "'");
+                    }
+                    paramSegments.add(bodyFieldValue);
+                    break;
+                default:
+                    break;
+            }
+        }
+        // Validate path param names and signature. Signature params must be a subset of path and body params.
```

Comment:
Comment should be "body param" as we only have one. Also I wonder if we can move the below bits to another function.